examples/motoko_examples/http_counter/src/main.ts

```diff
@@ -0,0 +1,164 @@
+import { Update, Opt, nat16, Variant, Query, nat8, Func, ic } from 'azle';
+import encodeUtf8 from 'encode-utf8';
+
+type StreamingCallbackHttpResponse = {
+    body: nat8[];
+    token: Opt<Token>;
+};
+
+type Token = {
+    // add whatever fields you'd like
+    arbitrary_data: string;
+};
+
+type CallbackStrategy = {
+    callback: Callback;
+    token: Token;
+};
+
+type Callback = Func<(t: Token) => Query<StreamingCallbackHttpResponse>>;
+
+type StreamingStrategy = Variant<{
+    Callback: CallbackStrategy;
+}>;
+
+type HeaderField = [string, string];
+
+type HttpResponse = {
+    status_code: nat16;
+    headers: HeaderField[];
+    body: nat8[];
+    streaming_strategy: Opt<StreamingStrategy>;
+    upgrade: Opt<boolean>;
+};
+
+type HttpRequest = {
+    method: string;
+    url: string;
+    headers: HeaderField[];
+    body: nat8[];
+};
+
+// Look at the stable examples
+// TODO what is the stable keyword?
+let counter = 0;
+
+function isGzip(x: HeaderField) : boolean {
+    return x[0].toLowerCase() === "accept-encoding" && x[1].toLowerCase().includes("gzip");
+}
+
+function encode(string: string): nat8[] {
+    return Array.from(new Uint8Array(encodeUtf8(string)))
+}
+
+export function http_request(req: HttpRequest) : Query<HttpResponse> {
+    if (req.method === 'GET') {
+        if (req.headers.find(isGzip) === undefined) {
+            if (req.url === '/stream') {
+                return {
+                    status_code: 200,
+                    headers: [ ['content-type', 'text/plain'] ],
+                    body: encode('Counter'),
+                    // TODO implement streaming_strategy when azle can handle callbacks correctly.
+                    streaming_strategy: {},
+                    upgrade: null
+                }
+            }
+            return {
+                status_code: 200,
+                headers: [ ['content-type', 'text/plain'] ],
+                body: encode(`Counter is ${counter}\n${req.url}\n`),
+                streaming_strategy: null,
+                upgrade: null
+            }
+        }
+        return {
+            status_code: 200,
+            headers: [ ['content-type', 'text/plain'], ['content-encoding', 'gzip'] ],
+            // body: '\1f\8b\08\00\98\02\1b\62\00\03\2b\2c\4d\2d\aa\e4\02\00\d6\80\2b\05\06\00\00\00',
+            // Currently there is an issue with azle being able to process the above string from the motoko example.
```

Comment:
If you could prefix these comments with `TODO`, that's a convention used throughout the codebase that'll make it easy to find these. A `TODO` comment is something that should be addressed, and isn't simply there for informational purposes.